src/spatial_reactionsystem.jl

```diff
@@ -0,0 +1,142 @@
+### Fetch Required Packages ###
+
+# Very WIP (wanted to do with {S,T} and stuff, but got various errors, and wasn't needed right now).
+### Spatial Reaction Structure. ###
+### Describing a spatial reaction that involves species from two neighbouring compartments. ###
+struct SpatialReaction
+    """The rate function (excluding mass action terms)."""
+    rate::Any
+    """Reaction substrates."""
+    substrates::Tuple{Vector,Vector}
+    """Reaction products."""
+    products::Tuple{Vector,Vector}
+
+    """
+    `false` (default) if `rate` should be multiplied by mass action terms to give the rate law.
+    `true` if `rate` represents the full reaction rate law.
+    """
+    only_use_rate::Bool
+
+    function SpatialReaction(rate,substrates,products;only_use_rate=false)
+        new(rate,substrates,products,only_use_rate)
+    end
+    
+end
+
+### Lattice Reaction Network Structure ###
+# Couples:
+# A network of compartments (a meta graph that can contain some additional infro for each compartment).
+# A reaction network (that is simulated within each compartment).
+# A set of spatial reactions (denoting interaction between comaprtments).
+struct LatticeReactionSystem # <: MT.AbstractTimeDependentSystem # Adding this part messes up show, disabling me from creating LRSs
+    """The reaction system expanded onto a graph."""
+    rs::ReactionSystem
+    """The original (non-spatial) reaction system that was expanded onto a graph."""
+    rs_base::ReactionSystem
+    """The spatial reactions defined between individual nodes."""
+    spatial_reactions::Vector{SpatialReaction}
+    """The graph on which the lattice is defined."""
+    lattice::MetaGraph
```

Comment:
Just use a regular graph? The metagraph is only used to store the initial values right? I think it would be better to use the normal Catalyst/ModelingToolkit mechanisms for storing initial values i.e. 
```julia
@variables t X(t)=[1,2,3]
# or
setdefaults!(rn, [:X => [1,2,3]])
```

 (i.e. just store the initial values vector, with the i'th entry corresponding to the value for the i'th node).

Reply:
Parameter values are also stored, and (and this is not super important) the names of the individual cells/compartments. If we can get away with your notation I agree this is better. We probably can. I want to have some chat to some people that actually work with spatial models and see how what is convenient in the set-up.

Reply:
I've publishing papers that simulate spatial stochastic models for ~20 years...